src/main/scala/is/hail/asm4s/package.scala

```diff
@@ -11,10 +11,13 @@ import org.objectweb.asm.tree._
 import scala.collection.generic.Growable
 import scala.language.implicitConversions
 import scala.reflect.ClassTag
+import scala.reflect
 
 package object asm4s {
 
-  trait TypeInfo[T] {
+  def typeInfo[T](implicit tti: TypeInfo[T]): TypeInfo[T] = tti
+
+  abstract class TypeInfo[T : ClassTag] {
```

Comment:
This somehow seems unnecessarily convoluted.  I don't think  you should have `T: ClassTag`, but just abstract `val clTag[T]`.  (But I wouldn't call it classTag because that conflicts with the symbol in reflect of the same name.)

Reply:
This approach takes advantage of implicit resolution so that we don't need to write

```scala
val clTag[Int] = classTag[Int]
```

in each TypeInfo.